src/game/stats.c

```diff
@@ -0,0 +1,198 @@
+
+                    case FT_LAVA:
+                        break;
+
+                    case FT_TRIGGER:
+                        trig_flags = *data;
+                        data++;
+                        trig_type = (type >> 8) & 0x3F;
+                        do {
+                            trigger = *data++;
+                            if (TRIG_BITS(trigger) == TO_SECRET) {
+                                int16_t number = trigger & VALUE_BITS;
+                                if (!(secrets & (1 << number))) {
+                                    secrets |= (1 << number);
+                                    m_LevelSecrets++;
+                                }
+                            }
+                            if (TRIG_BITS(trigger) != TO_OBJECT) {
+                                if (TRIG_BITS(trigger) == TO_CAMERA) {
+                                    trigger = *data++;
+                                }
+                            } else {
+                                int16_t idx = trigger & VALUE_BITS;
+                                ITEM_INFO *item = &g_Items[idx];
+
+                                // Add Pierre pickup and kills if oneshot
+                                if (item->object_number == O_PIERRE
+                                    && trig_flags & IF_ONESHOT
+                                    && !m_KillableItems[idx]) {
+                                    m_KillableItems[idx] = true;
+                                    m_LevelPickups += PIERRE_ITEMS;
+                                    m_LevelKillables += 1;
+                                }
+
+                                // Add killable if object triggered
+                                if (m_IfKillable[item->object_number]
+                                    && !m_KillableItems[idx]) {
+                                    m_KillableItems[idx] = true;
+                                    m_LevelKillables += 1;
+
+                                    // Add mercenary pickups
+                                    if (item->object_number == O_SKATEKID) {
+                                        m_LevelPickups += SKATEKID_ITEMS;
+                                    }
+                                    if (item->object_number == O_COWBOY) {
+                                        m_LevelPickups += COWBOY_ITEMS;
+                                    }
+                                    if (item->object_number == O_BALDY) {
+                                        m_LevelPickups += BALDY_ITEMS;
+                                    }
+                                }
+                            }
+                        } while (!(trigger & END_BIT));
+                        break;
+                    }
+                } while (!(type & END_BIT));
+            }
+        }
+    }
+}
+
+void Stats_Init()
+{
+    for (int i = 0; m_KillableObjs[i] != NO_ITEM; i++) {
+        m_IfKillable[m_KillableObjs[i]] = true;
+    }
+}
```

Comment:
I think this needs to be ran at the beginning of stats calculation, not once per game launch?

Reply:
The enemies that are killable never change, so it should only need run once on game launch.

Reply:
Could we remove `Stats_Init` and replace `m_IfKillable` with something like this?

```c
static bool Stats_IsObjectKillable(int32_t obj_num)
{
    for (int i = 0; m_KillableObjs[i] != NO_ITEM; i++) {
        if (m_KillableObjs[i] == obj_num) {
            return true;
        }
    }
    return false;
}

Reply:
We could I guess, that's how the old method was sort of done. This just does more work since every time you check an object, you have to iterate through the `m_KillableObjs` array.

Reply:
Yes, I'd prefer it this way, we do millions of iterations per second anyway. This is just one calculation.